### PR TITLE
refactor: clean up DID origins & only delegation owner can revoke delegation

### PR DESCRIPTION
**Pull request by weichweich**
## fixes 

- [x] properly benchmark DID origin checks
- [x] use paths for constants in the runtime
- [x] edit PR template to not mention old custom type  checks
- [x] only delegation owner can revoke the delegation (deposit owner can reclaim)

## Checklist:

- [ ] I have verified that the code works
  - [ ] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] This PR does not introduce new custom types
  - [ ] If not, I have opened a companion PR with the type changes in the [KILT types-definitions repository](https://github.com/KILTprotocol/type-definitions/pulls)
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Review by wischli — COMMENTED** [review]
A few very minor things.

File: pallets/ctype/src/mock.rs [hunk 1]
```diff
@@ -134,52 +35,168 @@ where
...
+	pub type Block = frame_system::mocking::MockBlock<Test>;
+
+	#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, TypeInfo, Default)]
+	pub struct DidIdentifier(AccountId32);
+
+	impl From<AccountId32> for DidIdentifier {
+		fn from(acc: AccountId32) -> Self {
+			DidIdentifier(acc)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/ctype/src/mock.rs:64]: yes i can move it to the support crate

File: support/src/mock.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+///! This module contains utilities for testing.
+
+/// This pallet only contains an origin which supports separated sender and
+/// subject.
+///
+/// WARNING: This is only used for testing!
```
[the remark this answers is not in the document]
Comment (weichweich) [reply support/src/mock.rs:24]: Ah no! This is a doc comment for the `mod` that should follow in the next line.

File: pallets/delegation/src/lib.rs [hunk 3]
```diff
@@ -576,17 +576,13 @@ pub mod pallet {
...
 
 			let delegation = DelegationNodes::<T>::get(&delegation_id).ok_or(Error::<T>::DelegationNotFound)?;
 
 			// Node can only be removed by owner of either the deposit or the node, not the
 			// parent or another ancestor
-			ensure!(
-				delegation.deposit.owner == sender || delegation.details.owner == invoker,
-				Error::<T>::UnauthorizedRemoval
-			);
+			ensure!(delegation.details.owner == invoker, Error::<T>::UnauthorizedRemoval);
```
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/delegation/src/lib.rs]: 04c43ea

File: pallets/delegation/src/benchmarking.rs [hunk 4]
```diff
@@ -265,22 +273,25 @@ benchmarks! {
...
+		let origin = <T as Config>::EnsureOrigin::generate_origin(sender, leaf_acc_id);
+	}: _<T::Origin>(origin, delegation_id, hierarchy_id, delegate_acc_id, perm, sig)
 	verify {
 		assert!(DelegationNodes::<T>::contains_key(delegation_id));
 	}
 
 	// worst case #1: revoke a child of the root delegation
 	// because all of its children have to be revoked
-	// complexitiy: O(h * c) with h = height of the delegation tree, c = max number of children in a level
+	// // complexitiy: O(h * c) with h = height of the delegation tree, c = max number of children in a level
```
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/delegation/src/benchmarking.rs]: 04c43ea

File: pallets/attestation/src/mock.rs [hunk 5]
```diff
@@ -240,34 +262,25 @@ pub(crate) mod runtime {
 		type MaxDelegatedAttestations = MaxDelegatedAttestations;
 	}
 
-	const ALICE_SEED: [u8; 32] = [0u8; 32];
-	const BOB_SEED: [u8; 32] = [1u8; 32];
+	pub(crate) const ACCOUNT_00: kilt_primitives::AccountId = kilt_primitives::AccountId::new([1u8; 32]);
```
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/attestation/src/mock.rs:251]: fd20652

File: pallets/ctype/src/mock.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/ctype/src/mock.rs]: 035d4b7

**Review by wischli — APPROVED** [review]
LGTM!

**Pull request merged**